### PR TITLE
change type definition for Object.create

### LINKED ISSUE #8872: Type definition for Object.create
Is there a reason why in `lib.d.ts`, `Object.create` is defined as following:
```typescript
create(o: any, properties?: PropertyDescriptorMap): any;
```
and not as following
```typescript
create<T>(o: T): T;
create(o: any, properties: PropertyDescriptorMap): any;
```
?

### PR DESCRIPTION
**Pull request by plantain-00**
Fixes #8872

### CODE REVIEW

File: src/lib/es5.d.ts [hunk 1]
```diff
@@ -139,9 +139,21 @@ interface ObjectConstructor {
     /**
       * Creates an object that has the specified prototype, and that optionally contains specified properties.
```
Comment (mhegazy) [inline src/lib/es5.d.ts]: i woudl update the comment here to be more specific.

```ts
/**
  * Creates an object that has null prototype.
  */
```

**Comment by mhegazy** [comment]
thanks. one comment about comment text. otherwise LGTM.

File: src/lib/es5.d.ts (hunk 1 above)
Comment (plantain-00) [inline src/lib/es5.d.ts]: done.

**Comment by mhegazy** [comment]
thanks!

**Comment by benjaminmillhouse** [comment]
Hello @mhegazy - I'm not seeing this definition change in my local version of Visual Studio Code.  Is there a specific way to pull the TypeScript update for this type definition?

**Comment by mhegazy** [comment]
If you are using latest VSCode, you should see the definition from TS 2.2: https://github.com/Microsoft/TypeScript/blob/release-2.2/lib/lib.d.ts#L163-L170

**Comment by benjaminmillhouse** [comment]
That's what I am seeing in my IntelliSense.
Did the create<T>(o: T): T; not make the cut then?

**Comment by mhegazy** [comment]
It was changes to avoid breaking changes introduced.

**Comment by Oikio** [comment]
Good day, I tried to find out why generic overload from #8872:
```ts
create<T>(o: T): T;
```

 was omitted in PR, but couldn't find any clue. Was it for a reason? Because it will nice to be able to cast a type in such manner and it will not introduce any breaking changes. Quick suggestion:
```ts
create<T>(o: T): T;
create<T>(o: T, properties: PropertyDescriptorMap & ThisType<T>): T; // not sure about ThisType though
```

**Pull request merged**